File: Lista 2 - Introdução à linguagem Python/src/craps.py
from random import randint


def main() -> None:
    dice1: int = randint(1, 6)
    dice2: int = randint(1, 6)
    points: int = dice1 + dice2

    print(f"Dice 1: {dice1}, Dice 2: {dice2}")
    print(f"Points: {points}")

    if points in [7, 11]:
        print("You won!")
        return

    if points in [2, 3, 12]:
        print("You lost.")
        return

    print("Next stage!")

    count: int = 0
    while True:
        count += 1

        dice1 = randint(1, 6)
        dice2 = randint(1, 6)
        sum_: int = dice1 + dice2 # "sum" is the name of a built-in function

        print(f"Dice 1: {dice1}, Dice 2: {dice2}")
        print(f"Play {count}: {sum_}")

        if sum_ == 7:
            print("You lost.")
            return

        if sum_ == points:
            print("You won!")
            return
            

if __name__ == "__main__":
    main()

File: Lista 1 - Introdução à linguagem Java/src/main/java/dev/italoseara/exercises/Craps.java
package dev.italoseara.exercises;

import java.util.List;

public class Craps {
    public static void main(String[] args) {
        int dice1 = rollDice();
        int dice2 = rollDice();
        int points = dice1 + dice2;

        System.out.printf("Dice 1: %d, Dice 2: %d\n", dice1, dice2);
        System.out.printf("Points: %d\n", points);

        if (List.of(7, 11).contains(points)) {
            System.out.println("You won!");
            return;
        }

        if (List.of(2, 3, 12).contains(points)) {
            System.out.println("You lost.");
            return;
        }

        System.out.println("Next stage!");

        int count = 0;
        while (true) {
            count++;

            dice1 = rollDice();
            dice2 = rollDice();
            int sum = dice1 + dice2;

            System.out.printf("Dice 1: %d, Dice 2: %d\n", dice1, dice2);
            System.out.printf("Play %d: %d\n", count, sum);

            if (sum == 7) {
                System.out.println("You lost.");
                return;
            }

            if (sum == points) {
                System.out.println("You won!");
                return;
            }
        }
    }

    private static int rollDice() {
        return (int) (Math.random() * 6) + 1;
    }
}
